### ISSUE TITLE
Material UI Virtuoso table with checkboxes (select/select all) causes undesired scroll to top 

### ISSUE THREAD

**Issue by maria-noaptes** [petyosi/react-virtuoso#862]
Hello! I'm trying to implement a table with rows that are selectable, with a checkbox for every row and  a <select all> checkbox in the header of the table.

I have an array state with the indexes that are selected in the component that is rendering the ReactVirtuoso table .
The problem is that every time I'm checking a checkbox the table scrolls to top. I guess it rerendering because the array prop is changing. 
When using just the Material UI table without virtuoso this is not happening.

Is there a workaround I can do to avoid this problem?

**Comment by maria-noaptes** [comment]
```import React, { useState } from "react";

import { Box, Table, TableBody, TableContainer, TableHead, TableRow } from "@mui/material";
import TableCell, { tableCellClasses } from "@mui/material/TableCell";

import CheckBoxForRow from "../atoms/TableAtoms/CheckBoxForRow";
import { TableVirtuoso } from "react-virtuoso";

export default function Page() {
  const [rowsSelected, setRowsSelected] = useState([]);

  const columnTitles = ...;
  const minimumToRenderVirtualizedTable = 10;

  const rows = [];
  data.forEach((elem) =>
      const row = {};
      // create a row component with TableCells
      rows.push(row);
    })
  );

  const VirtuosoTableComponents = {
    Scroller: React.forwardRef((props, ref) => <TableContainer {...props} ref={ref} />),
    Table: (props) => <Table {...props} sx={{ borderCollapse: "separate", tableLayout: "fixed" }} />,
    TableBody: React.forwardRef((props, ref) => <TableBody {...props} ref={ref} />),
  };

  function fixedHeaderContent() {
    return (
      <TableHead>
        // checkbox changes rowsSelected
       <CheckBoxForRow parent={true} rowsSelected={rowsSelected} setRowsSelected={setRowsSelected} totalRows={rows.length}> 
       </CheckBoxForRow>

        {columnTitles.map((column, index) => (
          <TableCell key={column} variant="head" align={index === 0 ? "left" : "center"}>
            {column}
          </TableCell>
        ))}
      </TableHead>
    );
  }

  function RowContent({ _index, row }) {
    return (
      <TableRow sx={styles["row"]}>
        <CheckBoxForRow parent={false} index={_index} rowsSelected={rowsSelected} setRowsSelected={setRowsSelected} totalRows={rows.length}></CheckBoxForRow>
        {columnTitles.map((column, index) => (
          <TableCell key={index} component={index === 0 ? "th" : ""} scope="row" align={index !== 0 ? "center" : "left"}>
            {row[column]}
          </TableCell>
        ))}
      </TableRow>
    );
  }
  const tableRows = rows.map((row, index) => <RowContent key={index} _index={index} row={row} totalRows={rows.length}></RowContent>);

  return (
        <>
          <TableVirtuoso
            data={rows}
            totalCount={minimumToRenderVirtualizedTable}
            style={{ height: 600 }}
            components={VirtuosoTableComponents}
            fixedHeaderContent={fixedHeaderContent}
            itemContent={(_index, row) => tableRows[_index]}
          />
       </>
   
}

```

**Comment by petyosi** [comment]
Read the note here: https://virtuoso.dev/customize-structure/

**Comment by maria-noaptes** [comment]
Hello! Thank you for the quick response and for the library!
 I saw the note there before but I don't know what context prop I should use. Is this something specific to Virtuoso?

Anyway, I managed to solve the problem by deleting this line
 `Scroller: React.forwardRef((props, ref) => <TableContainer {...props} ref={ref} />),`
which I used to give styles to the scroll bar, but it would be better to understand the solution with the context :)

Thank you again for your support.
